### PR TITLE
Device network initiated downlink interval

### LINKED ISSUE #3333: Make network-initiated downlink interval configurable
#### Summary

Make network-initiated downlink interval configurable

#### Why do we need this?

To allow for network-initiated downlink rate-limiting by the customer

#### What is already there? What do you see now?

Hardcoded value in NS (https://github.com/TheThingsNetwork/lorawan-stack/blob/11cf05ba98cf6df44f064658ea4393183b0e7126/pkg/networkserver/networkserver.go#L57)

#### What is missing? What do you want to see?

Make this value configurable

#### How do you propose to implement this?

Add this as field in device MAC settings

#### How do you propose to test this?

integration testing

#### Can you do this yourself and submit a Pull Request?

yes

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

Closes #3333.

#### Changes

- Add `network_initiated_downlink_interval` field to MAC settings (~do not use~ **use** gogoproto for duration, similar to other MAC settings, to avoid having to deal with *pbtypes.Duration in CLI command line flags).
- Use interval from MAC settings (if set), otherwise fallback to default value of 1 second.

#### Testing

:monkey:

##### Regressions

Changes are fully backwards compatible, existing devices have no interval set so will keep using the default value.

#### Notes for Reviewers

This issue is small enough, so worked on it to get it over with, since it requires some probably not too obvious changes in the `pkg/ttnpb` package for `FieldIsZero` and validations.

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [X] Documentation: Relevant documentation is added or updated.
- [X] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by adriansmares — APPROVED** [review]

File: api/end_device.proto
Comment (johanstokking) [inline api/end_device.proto]: I think we use `class_b_c` in other places; that would be instead of `network_initiated`. I know it's not consistent with NS terminology, but this is API so that should stay consistent with API, that is more important than introducing NS terms here.

**Comment by neoaggelos** [comment]
This needs code-owner approval and is good to merge.

File: api/end_device.proto
Comment (KrishnaIyer) [inline api/end_device.proto:273]: I'd reframe this as 
```
The minimum duration that must pass before a network-initiated (e.g. Class B or C) downlink is sent following an arbitrary downlink.
```
This feels like a `downlink spacing` or a `scheduling delay` rather than an `interval` but that's just be me.

Btw, what is an arbitrary downlink?

File: pkg/networkserver/mac/utils.go
Comment (KrishnaIyer) [inline pkg/networkserver/mac/utils.go:577]: For devices that don't have this field in the registry, does this just return nil and we skip to `L583`?

Or do existing devices need a migration to set this field in the DB and then set a default value? 

This is just for me since I don't know much about how non existent fields are handled when you get them from Redis. cc: @adriansmares

**Review by KrishnaIyer — COMMENTED** [review]
API and implementation look good to me. Small questions

File: pkg/networkserver/mac/utils.go
Comment (neoaggelos) [reply pkg/networkserver/mac/utils.go:577]: This is `nil` for existing devices, indeed, no migrations required.

Comment (KrishnaIyer) [reply pkg/networkserver/mac/utils.go:577]: Perfect thanks.

File: api/end_device.proto
Comment (adriansmares) [reply api/end_device.proto:273]: You have device initiated (class A) and network initiated (class B/C). Arbitrary here is meant to catch both.

Comment (KrishnaIyer) [reply api/end_device.proto:273]: Indeed but the word "arbitrary" is not defined in the LoRaWAN spec as a collective term and contrarily you cannot send an arbitrary downlink since their types are predefined (A/B/C). 
I'd have rephrased the following 
```
following an arbitrary downlink.
```
as
```
following a downlink.
```

But this is a minor point.

**Review by KrishnaIyer — APPROVED** [review]

**Pull request merged**
